Mathlib/Analysis/Matrix/Order.lean

```diff
@@ -0,0 +1,260 @@
+/-
+Copyright (c) 2025 Monica Omar. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Monica Omar
+-/
+import Mathlib.Analysis.CStarAlgebra.ContinuousFunctionalCalculus.Instances
+import Mathlib.Analysis.SpecialFunctions.ContinuousFunctionalCalculus.Rpow.Basic
+import Mathlib.LinearAlgebra.Matrix.HermitianFunctionalCalculus
+import Mathlib.LinearAlgebra.Matrix.PosDef
+
+/-!
+# The partial order on matrices
+
+-/
+
+variable {𝕜 n : Type*} [RCLike 𝕜] [Fintype n]
+
+open scoped ComplexOrder
+open Matrix
+
+namespace Matrix
+
+section PartialOrder
+variable {R : Type*} [Ring R] [StarRing R] [PartialOrder R] [AddLeftMono R]
+
+/-- The preorder on matrices given by `A ≤ B := (B - A).PosSemidef`. -/
+abbrev instPreOrder [AddLeftMono R] : Preorder (Matrix n n R) where
+  le A B := (B - A).PosSemidef
+  le_refl A := sub_self A ▸ PosSemidef.zero
+  le_trans A B C h₁ h₂ := sub_add_sub_cancel C B A ▸ h₂.add h₁
+
+scoped[MatrixOrder] attribute [instance] Matrix.instPreOrder
+
+open MatrixOrder
+
+lemma le_iff {A B : Matrix n n R} : A ≤ B ↔ (B - A).PosSemidef := Iff.rfl
+
+lemma nonneg_iff {A : Matrix n n R} : 0 ≤ A ↔ A.PosSemidef := by rw [le_iff, sub_zero]
+
+protected alias ⟨LE.le.posSemidef, PosSemidef.nonneg⟩ := nonneg_iff
+
+attribute [aesop 20% apply (rule_sets := [CStarAlgebra])] PosSemidef.nonneg
+
+/-- The partial order on matrices given by `A ≤ B := (B - A).PosSemidef`. -/
+abbrev instPartialOrder : PartialOrder (Matrix n n 𝕜) where
+  le_antisymm A B h₁ h₂ := by
+    have foo := neg_nonneg.mp <| trace_neg (A - B) ▸ neg_sub A B ▸ h₁.trace_nonneg
+    have : (A - B).trace = 0 := le_antisymm foo h₂.trace_nonneg
+    classical
+    simp_rw [h₂.isHermitian.trace_eq_sum_eigenvalues, ← RCLike.ofReal_sum,
+      RCLike.ofReal_eq_zero, Finset.sum_eq_zero_iff_of_nonneg (s := Finset.univ)
+        (by simpa using h₂.eigenvalues_nonneg), Finset.mem_univ, true_imp_iff] at this
+    exact sub_eq_zero.mp <| funext_iff.eq ▸ h₂.isHermitian.eigenvalues_eq_zero_iff.mp <| this
```

Comment:
I think you should extract this out into a lemma (which should go in the other file) that the trace is faithful.